tools/e2e/commands.ts

```diff
@@ -156,9 +157,31 @@ Cypress.Commands.add('clearRecordDrafts', () => {
 })
 
 Cypress.Commands.add('editor_readFormUniqueIdentifier', () => {
-  return cy
-    .get('gn-ui-form-field[ng-reflect-model=uniqueIdentifier] input')
-    .invoke('val')
+  cy.url().then((url) => {
+    if (url.includes('/edit/')) {
+      return url.split('edit/').pop()
+    }
+  })
+})
+
+Cypress.Commands.add('editor_findDraftInLocalStorage', () => {
+  cy.window().then((win) => {
+    return new Cypress.Promise((resolve, reject) => {
+      const checkLocalStorage = () => {
+        const keys = Object.keys(win.localStorage)
+        const matchingKey = keys.find((key) =>
+          key.startsWith('geonetwork-ui-draft-')
+        )
+
+        if (matchingKey) {
+          resolve(win.localStorage.getItem(matchingKey))
+        } else {
+          setTimeout(checkLocalStorage, 100) // Retry after 100ms
```

Comment:
Maybe we could set a limited about of retries then? Because this fails 90% of the time without this retrying workflow, the draft saving being too slow compared to the speed of the tests, the localStorage often doesn't have the draft in it yet.

Reply:
Ah ok I see... maybe in this case we can leave it like it is. I mean this command will only be used if there is actually a draft present.
@jahow @LHBruneton-C2C any opinions on that?

Reply:
I didn't follow all the details, but I do have some insight. I was faced with the same issue recently, and did this (already committed):
```
// wait for the draft to be saved
  cy.get('@recordUuid').then((recordUuid) => {
    // nesting thens as Cypress doesn't seem to support the "all" operator
    //https://github.com/cypress-io/cypress/issues/915
    cy.get('@previousDraft').then((previousDraft) => {
      cy.window()
        .its('localStorage')
        .invoke('getItem', `geonetwork-ui-draft-${recordUuid}`)
        .should('not.eq', previousDraft)
    })
  })
```
Here, nesting the gets means it's Cypress that will deal with fails and retries.

Reply:
I think the detail you're missing it that it's now impossible to get the recordUuid from the DOM, since it's not going to be displayed. So your solution is actually what I'm trying to change :smile:

Reply:
I was only suggesting how not to deal with retries ;)

Reply:
Okay! Took me a bit of time but I think I understood your point (in my last commit) :)